src/lib.rs

```diff
@@ -1692,7 +1692,7 @@ pub trait ByteOrder:
     #[inline]
     fn from_slice_i16(src: &mut [i16]) {
         let src = unsafe {
-            slice::from_raw_parts_mut(src.as_ptr() as *mut u16, src.len())
+            slice::from_raw_parts_mut(src.as_mut_ptr() as *mut u16, src.len())
```

Comment:
Is the `as ...` even still needed? I think you can make the code slightly more type-safe by removing the casts. :D

(Raw-ptr-casts always make me nervous as they are so incredibly unchecked.)

Reply:
Ah, right... If you like, you could use [the `cast` method](https://doc.rust-lang.org/nightly/std/primitive.pointer.html#method.cast) which makes sure that only the pointee type is changed (but mutability is not).